tests/osaka/eip7939_count_leading_zeros/test_count_leading_zeros.py

```diff
@@ -6,80 +6,219 @@
 
 
 @pytest.mark.valid_from("Osaka")
-@pytest.mark.parametrize("bits", range(257))
-def test_clz_filled_format_0b0111(state_test: StateTestFiller, bits: int, pre: Alloc, fork: Fork):
+@pytest.mark.parametrize(
+    "test_id,value,expected_clz",
+    clz_parameters(),
+    ids=[f"{test_data[0]}-expected_clz_{test_data[2]}" for test_data in clz_parameters()],
+)
+def test_clz_opcode_scenarios(
+    state_test: StateTestFiller,
+    pre: Alloc,
+    test_id: str,
+    value: int,
+    expected_clz: int,
+):
     """
-    Test CLZ opcode. Value is of format 0b0011..11.
-    The CLZ(0) case is also included in this test.
+    Test CLZ opcode functionality.
+
+    Cases:
+    - Format 0xb000...111: leading zeros followed by ones (2**256 - 1 >> bits)
+    - Format 0xb010...000: single bit set at position (1 << bits)
+
+    Test coverage:
+    - Leading zeros pattern: 0b000...111 (0 to 256 leading zeros)
+    - Single bit pattern: 0b010...000 (bit at each possible position)
+    - Edge cases: CLZ(0) = 256, CLZ(2^256-1) = 0
     """
-    value = 2**256 - 1 >> bits
+    sender = pre.fund_eoa()
     contract_address = pre.deploy_contract(
-        Op.SSTORE(0, Op.CLZ(value)),
+        code=Op.SSTORE(0, Op.CLZ(value)),
         storage={"0x00": "0xdeadbeef"},
     )
-    sender = pre.fund_eoa()
-
-    tx = Transaction(to=contract_address, sender=sender, gas_limit=200_000)
-
+    tx = Transaction(
+        to=contract_address,
+        sender=sender,
+        gas_limit=200_000,
+    )
     post = {
-        # gas cost calculation adds the cost of PUSH (3), POP (2) and GAS (2) to the costs of CLZ
-        # TODO: read gas costs of these opcode from fork
-        contract_address: Account(storage={"0x00": bits}),
+        contract_address: Account(storage={"0x00": expected_clz}),
     }
-
     state_test(pre=pre, post=post, tx=tx)
 
 
 @pytest.mark.valid_from("Osaka")
-@pytest.mark.parametrize("bits", range(0, 256))
-def test_clz_filled_format_0b0100(state_test: StateTestFiller, bits: int, pre: Alloc, fork: Fork):
-    """Test CLZ opcode. Value is of format 0b0010..00."""
-    value = 1 << bits
+def test_clz_gas(state_test: StateTestFiller, pre: Alloc, fork: Fork):
+    """Test CLZ opcode gas cost."""
     contract_address = pre.deploy_contract(
-        Op.SSTORE(0, Op.CLZ(value)),
+        Op.SSTORE(
+            0,
+            CodeGasMeasure(code=Op.CLZ(Op.PUSH1(1)), extra_stack_items=1, overhead_cost=0),
+        ),
         storage={"0x00": "0xdeadbeef"},
     )
     sender = pre.fund_eoa()
-
     tx = Transaction(to=contract_address, sender=sender, gas_limit=200_000)
-
     post = {
-        # gas cost calculation adds the cost of PUSH (3), POP (2) and GAS (2) to the costs of CLZ
-        # TODO: read gas costs of these opcode from fork
-        contract_address: Account(storage={"0x00": 255 - bits}),
+        contract_address: Account(  # Cost measured is CLZ + PUSH1
+            storage={"0x00": Spec.CLZ_GAS_COST + fork.gas_costs().G_VERY_LOW}
```

Comment:
I feel like we should simply use `fork.gas_costs().G_VERY_LOW` again here, because otherwise we are deviating from the way we usually measure gas and link gas expectations to the fork.
```suggestion
            storage={"0x00": fork.gas_costs().G_VERY_LOW}
```